### DISCUSSION TITLE
Get User-Friendly Addresses in Ton Blockchain

### DISCUSSION THREAD

**Discussion by xxargonaut** [community/community#135413] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Addresses in TON can be represented in two formats Raw and User-Friendly Addresses, sometimes you will need to convert them from one form to another. How to convert the address a3935861f79daf59a13d6d182e1640210c02f98e3df18fda74b8f5ab141abf18 in workchain: 0 in friendly format using the ton-core library?

**Comment by 0xpuck** [comment, marked as answer]
This is an interesting problem! If you're working with address conversions in TON, I'd recommend checking out the ton-core library's documentation. It has some handy functions for managing different address formats.

- Let's use the `Address` class defined in `@ton/core` to parse and convert raw & user-friendly addresses from
- Create the Address object using the workchain and raw address   `const address: Address = new Address(workChain: number, hash: Buffer);`   At this time, Buffer is obtained as follows: `Buffer.from(rawAddress, 'hex')`
- Call **toString()** on an address instance to convert it into a user-friendly address string. Here's how to use **toString():** `address.toString({ testOnly: boolean, bounceable: boolean });`

This is full code:
```
const { Address } = require('@ton/core');

const rawAddress = 'a3935861f79daf59a13d6d182e1640210c02f98e3df18fda74b8f5ab141abf18';
const workchain = 0;

const address = new Address(workchain, Buffer.from(rawAddress, 'hex'));

const friendlyAddress = address.toString({
  testOnly: false,
  bounceable: true,
});

console.log(friendlyAddress);
```

**Reply by 0xIntelTech** [reply]
do you know about Ton?

**Reply by xxargonaut** [reply]
I think you are skilled developer

**Reply by 0xpuck** [reply]
Thank you
I just know a little bit about Ton Blockchain as a blockchain developer.
Please feel free to ask questions about any issues that arise in the future.
I will tell you as much as I know.

**Reply by xxargonaut** [reply]
I ran the code and got the result.

`EQCjk1hh952vWaE9bRguFkAhDAL5jj3xj9p0uPWrFBq_GEMS`

That's correct.
thank you

**Comment by DevOgabek** [comment]
how i can do this with python?

**Comment by Ramengel** [comment]
I need some help
My friendly address in @tonconnect/react-ui and in tonkeeper are the same
But on backend when I transfer raw Address with  { Address } @ton/core to user Friendly is not the same
Am I doing smth wrong ?

![Screenshot 2024-10-04 081637](https://github.com/user-attachments/assets/5bc0ce46-b9fe-4af7-9c09-8075245d9711)
